lib/spack/spack/database.py

```diff
@@ -0,0 +1,290 @@
+##############################################################################
+# Copyright (c) 2013, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Written by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://scalability-llnl.github.io/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License (as published by
+# the Free Software Foundation) version 2.1 dated February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os
+import sys
+import inspect
+import glob
+import imp
+
+import time
+import copy
+import errno
+
+from external import yaml
+from external.yaml.error import MarkedYAMLError
+
+import llnl.util.tty as tty
+from llnl.util.filesystem import join_path
+from llnl.util.lang import *
+
+import spack.error
+import spack.spec
+from spack.spec import Spec
+from spack.error import SpackError
+from spack.virtual import ProviderIndex
+from spack.util.naming import mod_to_class, validate_module_name
+
+
+def _autospec(function):
```

Comment:
I was also just starting to look at this.  Is there a way to get my 
existing builds into a database?  After updating my existing repo 'spack 
find' reports 0 packages installed.

-Matt

On Wed, 26 Aug 2015, Todd Gamblin wrote:
> 
> In lib/spack/spack/database.py:
> 
> > +from external import yaml
> > +from external.yaml.error import MarkedYAMLError
> > +
> > +import llnl.util.tty as tty
> > +from llnl.util.filesystem import join_path
> > +from llnl.util.lang import *
> > +
> > +import spack.error
> > +import spack.spec
> > +from spack.spec import Spec
> > +from spack.error import SpackError
> > +from spack.virtual import ProviderIndex
> > +from spack.util.naming import mod_to_class, validate_module_name
> > +
> > +
> > +def _autospec(function):
> 
> Should probably consolidate the various autospec decorators at some point,
> but this is fine for now. There are getting to be a lot of them in various
> files, and they do the same thing.
> 
> —
> Reply to this email directly or view it on
> GitHub.[AEO__pHddZ6xnpRk69FaD5rzohQAiVYiks5orfDkgaJpZM4FyKgl.gif]
> 
> 
>